sbol_utilities/build_planning.py

```diff
@@ -0,0 +1,142 @@
+import sbol3
+import tyto
+
+from sbol_utilities.component import get_subcomponents, get_subcomponents_by_identity
+
+# TODO: Change SBOL_ASSEMBLY_PLAN and sbol3.SBOL_DESIGN to tyto calls after resolution of 
+SBOL_ASSEMBLY_PLAN = 'http://sbols.org/v3#assemblyPlan'
+ASSEMBLY_TYPES = {sbol3.SBOL_DESIGN, SBOL_ASSEMBLY_PLAN}
+
+
+def validate_part_in_backbone(pib: sbol3.Component) -> bool:
+    """Check if a Component represents a part in backbone
+
+    :param plan: Component being validated
+    :return: true if its structure follows the best practices for representing a part in backbone
+    """
+    subcomps = get_subcomponents(pib)
+
+    has_insert = False
+    has_backbone = False
+
+    i = 0
+    while (not has_insert or not has_backbone) and i < len(subcomps):
+        part = subcomps[i].instance_of.lookup()
+        
+        if tyto.SO.engineered_insert in subcomps[i].roles or tyto.SO.engineered_insert in part.roles:
+            has_insert = True
+        elif is_backbone(part):
+            has_backbone = True
+            
+        i = i + 1
+
+    return has_insert and has_backbone
+
+
+def is_backbone(b: sbol3.Component) -> bool:
+    """Check if Component is a backbone
+
+    :param plan: Component being checked
+    :return: true if it has an expected role for a backbone
+    """
+    for role in b.roles:
+        if role == tyto.SO.vector_replicon or tyto.SO.vector_replicon.is_ancestor_of(role):
+            return True
+
+    return False
+
+
+def validate_composite_part_assemblies(c: sbol3.Component) -> bool:
+    """Check if a Component for a composite part has valid assemblies
```

Comment:
Same as above. Please explain what the function actually does in each of the functions here and below.